batch2/batch/google_compute.py

```diff
@@ -17,34 +17,43 @@
 
 
 class EntryIterator:
-    def __init__(self, gservices):
+    def __init__(self, gservices, db):
         self.gservices = gservices
-        self.mark = time.time()
+        self.db = db
+        self.mark = None
         self.entries = None
-        self.latest = None
+
+    async def async_init(self):
+        row = await self.db.execute_and_fetchone('SELECT * FROM `gevents_mark`;')
+        if row['mark']:
+            self.mark = row['mark']
+        else:
+            now = datetime.datetime.utcnow().isoformat() + 'Z'
+            await self._update_mark(now)
+
+    async def _update_mark(self, timestamp):
+        await self.db.execute_update(
+            'UPDATE `gevents_mark` SET mark = %s;',
+            (timestamp,))
+        self.mark = timestamp
 
     def __aiter__(self):
         return self
 
     async def __anext__(self):
         while True:
             if not self.entries:
-                self.entries = await self.gservices.list_entries()
+                self.entries = await self.gservices.list_entries(self.mark)
+            timestamp = None
             try:
                 entry = await anext(self.entries)
-                timestamp = entry.timestamp.timestamp()
-                if not self.latest:
-                    self.latest = timestamp
-                # might miss events with duplicate times
-                # solution is to track resourceId
-                if timestamp <= self.mark:
-                    raise StopAsyncIteration
+                timestamp = entry.timestamp.isoformat() + 'Z'
```

Comment:
You're sure the default order is ascending (older to newer)?

Reply:
I'll add the ordering back to make sure.